### DISCUSSION TITLE
API call flow when switching between anonymous and signed in users

### DISCUSSION THREAD

**Discussion by apfelfabrik** [Iterable/react-native-sdk#531] [Q&A]
Hey everyone,

We're allowing users to open our app and sign in optionally at a later point in time. What seems to work well is the push notification device registration for anonymous users using a JWT-less API token when we initialize the SDK. Now it seems we're running into some race conditions when the user signs up/in. We need to unregister the anonymous user's device from push notifications, we need to initialize the SDK again with the JWT-based API token, and we need to register the device again for the now authenticated user. 

**What would be the correct or recommended order of react native SDK API calls to achieve this, including calls to `initialize` with the individual API keys?**

Please excuse if this is a well-documented case, I couldn't find anything on this in the docs even though I think this should be rather common practice considering user management in mobile apps.

Cheers,
Martin.

**Comment by lposen** [comment]
Hi @apfelfabrik,

We have not yet implemented JWT for RN.  However, the implementation is currently in development.

I will reach back out to you:
1. once JWT implementation has been released, and
2. once I have consulted with the team and figured out the correct order of operations for the scenario you describe.

I will take a look through our documentation as well.  The flow you describe doesn't sound like an uncommon one, so it should likely be documented.

If you would like to partner with us on this, please contact your CSM!

**Comment by lposen** [comment]
Hi @apfelfabrik and @dazlious, 

Wanted to circle back let you know that JWT handling is currently in development, and we plan on releasing it before the end of Q4.

**Comment by lposen** [comment]
Hi all!  JWT has been released as of version 2.2.0.  Let me know if you run into any further issues.
